migration/20170713-17-move-externalintegration-columns-to-settings.sql

```diff
@@ -18,4 +18,22 @@ WHERE e.password is not null;
 
 ALTER TABLE externalintegrations DROP COLUMN url;
 ALTER TABLE externalintegrations DROP COLUMN username;
-ALTER TABLE externalintegrations DROP COLUMN password;
\ No newline at end of file
+ALTER TABLE externalintegrations DROP COLUMN password;
+
+-- Add the collection protocols to the external integrations.
+UPDATE externalintegrations as e
+SET protocol = c.protocol
+FROM collections as c
+WHERE e.id = c.external_integration_id;
+
+-- Create an externalintegration for the Open Access Content Server.
+INSERT INTO externalintegrations(protocol)
+SELECT c.protocol
+FROM collections c
+WHERE c.external_integration_id is null;
+
```

Comment:
With the changes to Collection on master, collections and their external integrations are created before ExternalIntegration.protocol exists. At that point, external integrations are also only created for a Collection if it has a url, username, or password.

Content Server doesn't have any of those attributes, so it's never associated with an external integration in the database. Because the oversight should only occur for the content server at this point, this migration creates an integration for any collection that doesn't have one (content server, if anything) and then associates that collection with the new integration below.